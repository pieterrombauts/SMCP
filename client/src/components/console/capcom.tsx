import React from 'react';
import 'bootstrap/dist/css/bootstrap.min.css';
import Tabs from 'react-bootstrap/Tabs'
import Tab from 'react-bootstrap/Tab'
import capcom_1 from "./../../media/Capcom_1.png"

interface AppProps {};

const capcom: React.FC<AppProps> = () => {
  return(
         <Tabs defaultActiveKey="Spartan_console" id="uncontrolled-tab-example" style={{ float: "right"}} >
        
          <Tab eventKey="CAPCOM_1" title="Script">
            <img src={capcom_1} style={{width:"300px"}}/> 
          </Tab>

         
        </Tabs>
  );
}

export const Capcom = capcom;