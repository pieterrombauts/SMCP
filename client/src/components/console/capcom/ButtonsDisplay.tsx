import React, {useState, useEffect} from 'react';
import styled from 'styled-components';
import moment from 'moment';

interface ButtonsProps {
  className?: string;
  time: string;
};

const UButtonsDisplay: React.FC<ButtonsProps> = ( props ) => {
  const [ signal, setSignal ] = useState<boolean>(true)
  const [ losTimes, setLosTimes ] = useState<string[]>(['2019-11-15T15:17:37', '2019-11-15T15:57:51', '2019-11-15T16:03:53', '2019-11-15T16:07:03', '2019-11-15T16:10:10', '2019-11-15T16:13:24','2019-11-15T16:52:26','2019-11-15T17:37:00'])
  const [ aosTimes, setAosTimes ] = useState<string[]>(['2019-11-15T15:17:52', '2019-11-15T15:59:28', '2019-11-15T16:05:46', '2019-11-15T16:09:19', '2019-11-15T16:12:07', '2019-11-15T16:14:25','2019-11-15T17:00:00','2019-11-15T17:44:00'])
  const [ nextAOS, setNextAOS ] = useState<string>("");
  const [ nextLOS, setNextLOS ] = useState<string>("");
  useEffect(() => {
    if (moment(props.time).isBefore(moment(aosTimes[0]))) {
      var hoursAOS = moment(aosTimes[0]).diff(moment(props.time), 'hours')
      var minutesAOS = moment(aosTimes[0]).diff(moment(props.time), 'minutes')
      var secondsAOS = moment(aosTimes[0]).diff(moment(props.time), 'seconds')
      setNextAOS(hoursAOS.toString().padStart(2,'0') + ":" + (minutesAOS - 60*hoursAOS).toString().padStart(2,'0') + ":" + (secondsAOS - 60*minutesAOS).toString().padStart(2,'0'));
    } else if (moment(props.time).isSameOrAfter(moment(aosTimes[0]))) {
      setSignal(true);
      var newAosTimes = aosTimes;
      newAosTimes.shift();
      setAosTimes(newAosTimes);
    }
    if (moment(props.time).isBefore(moment(losTimes[0]))) {
      var hoursLOS = moment(losTimes[0]).diff(moment(props.time), 'hours')
      var minutesLOS = moment(losTimes[0]).diff(moment(props.time), 'minutes')
      var secondsLOS = moment(losTimes[0]).diff(moment(props.time), 'seconds')
      setNextLOS(hoursLOS.toString().padStart(2,'0') + ":" + (minutesLOS - 60*hoursLOS).toString().padStart(2,'0') + ":" + (secondsLOS - 60*minutesLOS).toString().padStart(2,'0'));
    } else if (moment(props.time).isSameOrAfter(moment(losTimes[0]))) {
      setSignal(false);
      var newLosTimes = losTimes;
      newLosTimes.shift();
      setLosTimes(newLosTimes);
    }
  }, [props.time, signal, losTimes, aosTimes])
  return(
    <div className={props.className}>
      <svg preserveAspectRatio='xMidYMid meet' viewBox='0 0 1300 990'>
        <rect x="740" y="20" rx="5" ry="5" width="560" height="400" style={{fill:"#393939", stroke:"#393939", strokeWidth:5, opacity:0.5}} />
        

        
        <rect x="770" y="40" rx="5" ry="5" width="200" height="100" style={{fill: signal ? "#00CC00" : "#004000", stroke:"#393939", strokeWidth:5}} />
        
        <text x="810" y="110" fill={signal ? "white" : "grey"} fontSize="50" fontFamily="Verdana">AOS</text>

        <rect x="1060" y="40" rx="5" ry="5" width="200" height="100" style={{fill: signal ? "#640000" : "#FF0000", stroke:"#393939", strokeWidth:5}} />
        <text x="1110" y="110" fill={signal ? "grey" : "white"} fontSize="50" fontFamily="Verdana">LOS</text>

        <text x="780" y="230" fill="white" fontSize="50" fontFamily="Verdana">Next AOS {nextAOS}</text>
        <text x="780" y="310" fill="white" fontSize="50" fontFamily="Verdana">Next LOS {nextLOS}</text>

        <rect x='0' y='20' rx="5" ry="5" width="700" height="400" style={{fill:"#333333", stroke:"#393939", strokeWidth:5, opacity:0.5}} />
        
        <text x="30" y="100" fill="white" fontSize="50" fontFamily="Verdana" text-decoration="underline">Caution & Warning </text>
        
        <rect x="20" y="140" rx="5" ry="5" width="110" height="90" style={{fill:"white", stroke:"#393939", strokeWidth:5}} />
        <text x="40" y="195" fill="black" fontSize="30" fontFamily="Verdana">FIRE</text>

        <rect x="160" y="140" rx="5" ry="5" width="110" height="90" style={{fill:"white", stroke:"#393939", strokeWidth:5}} />
        <text x="180" y="195" fill="black" fontSize="30" fontFamily="Verdana">Î”P</text>

        <rect x="300" y="140" rx="5" ry="5" width="110" height="90" style={{fill:"white", stroke:"#393939", strokeWidth:5}} />
        <text x="320" y="195" fill="black" fontSize="30" fontFamily="Verdana">ATM</text>

        <rect x="440" y="140" rx="5" ry="5" width="110" height="90" style={{fill:"white", stroke:"#393939", strokeWidth:5}} />
        <text x="450" y="195" fill="black" fontSize="18" fontFamily="Verdana">WARNING</text>

        <rect x="580" y="140" rx="5" ry="5" width="110" height="90" style={{fill:"white", stroke:"#393939", strokeWidth:5}} />
        <text x="590" y="195" fill="black" fontSize="18" fontFamily="Verdana">CAUTION</text>

        <rect x="580" y="300" rx="5" ry="5" width="110" height="90" style={{fill:"white", stroke:"#393939", strokeWidth:5}} />
        <text x="595" y="355" fill="black" fontSize="30" fontFamily="Verdana">TEST</text>
      </svg>
    </div>
  )
};

const ButtonsDisplay = styled(UButtonsDisplay)`
  svg {
    height: 100%;
    position: absolute;
    left: 0;
  }
`;

export default ButtonsDisplay;