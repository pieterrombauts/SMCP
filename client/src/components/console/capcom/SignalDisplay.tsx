import React from 'react';
import styled from 'styled-components';

interface SignalProps {
  className?: string;
};

const USignalDisplay: React.FC<SignalProps> = ( props ) => {
  return(
    <div className={props.className}>
      <svg preserveAspectRatio='xMidYMid meet' viewBox='0 0 1300 990'>
      <rect x="740" y="20" rx="5" ry="5" width="560" height="400" style={{fill:"#393939", stroke:"#393939", strokeWidth:5, opacity:0.5}} />
      
      <rect x="770" y="40" rx="5" ry="5" width="200" height="100" style={{fill:"green", stroke:"#393939", strokeWidth:5}} />
      <text x="810" y="110" fill="white" fontSize="50" fontFamily="Verdana">AOS</text>

      <rect x="1060" y="40" rx="5" ry="5" width="200" height="100" style={{fill:"red", stroke:"#393939", strokeWidth:5}} />
      <text x="1110" y="110" fill="white" fontSize="50" fontFamily="Verdana">LOS</text>

      <text x="780" y="230" fill="white" fontSize="50" fontFamily="Verdana">Next AOS 0:02:34</text>
      <text x="780" y="310" fill="white" fontSize="50" fontFamily="Verdana">Next LOS 0:05:54</text>
      
      </svg>
    </div>
  )
};

const SignalDisplay = styled(USignalDisplay)`
  svg {
    width: 100%;
    height: 100%;
  }
`;

export default SignalDisplay;