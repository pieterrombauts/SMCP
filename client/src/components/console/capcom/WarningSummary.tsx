import React, { useState, useEffect } from 'react'
import Table from 'react-bootstrap/Table'
import styled from 'styled-components';
import { tableEntry } from '../customTypes'
import { RootState } from '../../../reducers';
import { connect, ConnectedProps } from 'react-redux';
import moment from 'moment';

interface warningSystemProps {
    className?: string;
};

const mapState = (state: RootState ) => ({
    time: state.dataReducer.time
})
  
const connector = connect(mapState, {})
type PropsFromRedux = ConnectedProps<typeof connector>
type Props = PropsFromRedux & warningSystemProps

const tableUpdateTimes: string[] = ['2019-11-15T15:04:00',
                                    '2019-11-15T15:07:02',
                                    '2019-11-15T15:11:04',
                                    '2019-11-15T15:14:06',
                                    '2019-11-15T15:20:08',
                                    '2019-11-15T15:23:10',
                                    '2019-11-15T15:31:12',
                                    '2019-11-15T15:37:14',
                                    '2019-11-15T15:40:16',
                                    '2019-11-15T15:47:18',
                                    '2019-11-15T15:51:20',
                                    '2019-11-15T15:58:22',
                                    '2019-11-15T16:02:24',
                                    '2019-11-15T16:10:26',
                                    '2019-11-15T16:17:28',
                                    '2019-11-15T16:23:30',
                                    '2019-11-15T16:27:32'];

const pendingTableEntries: tableEntry[] = [
    {eventNum: "9413", annun: "SUPP", cl: "C", ack: false, sys: "TCS", message: "Thermal Safing Load Shed Inhibited - LAB"},
    {eventNum: "12288", annun: "SUPP", cl: "C", ack: false, sys: "TCS", message: "Thermal Thermal Safing Node 2 LTL Load Shed Timer Started - Node2"},
    {eventNum: "13463", annun: "SUPP", cl: "C", ack: false, sys: "CDH", message: "DMS Mission Management Computer (MMC) Failure - COL"},
    {eventNum: "13598", annun: "ENA", cl: "C", ack: false, sys: "EPS", message: "Power Distribution Unit (PDU2) Output Overcurrent Trip - COL"},
    {eventNum: "13568", annun: "SUPP", cl: "C", ack: false, sys: "CDH", message: "DMS System Bus Failure - COL"},
    {eventNum: "13578", annun: "ENA", cl: "C", ack: false, sys: "EPS", message: "Power Distribution Unit (PDU1) Nominal Controller Failure - COL"},
    {eventNum: "12592", annun: "SUPP", cl: "C", ack: false, sys: "ECL", message: "Cabin Smoke Detector 2 Fail - COL"},
    {eventNum: "5049", annun: "SUPP", cl: "C", ack: false, sys: "TCS", message: "Thermal Saffing Partial LTL Load Shed Timer Started"},
    {eventNum: "5050", annun: "SUPP", cl: "C", ack: false, sys: "TCS", message: "Thermal Saffing Partial MTL Load Shed Timer Started"},
    {eventNum: "13529", annun: "SUPP", cl: "C", ack: false, sys: "ECL", message: "ppO2 Sensor 2 Low - COL"},
    {eventNum: "13558", annun: "SUPP", cl: "C", ack: false, sys: "ECL", message: "Loss of IMV Supply Function - COL"},
    {eventNum: "4173", annun: "ENA", cl: "C", ack: false, sys: "TCS", message: "ETCS Loop B PCVP Loss of Comm - P1"},
    {eventNum: "13535", annun: "SUPP", cl: "W", ack: false, sys: "TCS", message: "Cooling Loop Delta Pressure Sensor 2 Low - COL"},
    {eventNum: "10472", annun: "ENA", cl: "C", ack: true, sys: "TCS", message: "ETCS Loop B Radiator Environment Temp Too Cold - P1"},
    {eventNum: "4248", annun: "ENA", cl: "C", ack: true, sys: "TCS", message: "ETCS Loop B TRRJ Switchover Sequence Failure - P1"},
    {eventNum: "5100", annun: "SUPP", cl: "C", ack: true, sys: "CNT", message: "Loss of Active IAC Handover to Backup IAC"},
    {eventNum: "11099", annun: "ENA", cl: "C", ack: true, sys: "EPS", message: "RPCM S32B_A Observed vs Last Commanded State Discrepancy - S3"}
];

const UWarningSummary: React.FC<Props> = (props) => {
    const [ tableEntries, setTableEntries ] = useState<tableEntry[]>([]);
    useEffect(() => {
        if (moment.utc(props.time).isAfter(moment.utc(tableUpdateTimes[tableEntries.length]))) {
            setTableEntries([...tableEntries, pendingTableEntries[0]]);
            pendingTableEntries.shift();
        }
    }, [tableEntries, pendingTableEntries, props.time])
    
    return (
        <div className={props.className}>
            <Table bordered hover variant ="dark" size="sm">
                <thead>
                    <tr className="white">
                        <th style={{width: '100px'}}>EVENT#</th>
                        <th style={{width: '100px'}}>ANNUN</th>
                        <th style={{width: '100px'}}>CL</th>
                        <th style={{width: '100px'}}>ACK</th>
                        <th style={{width: '100px'}}>SYS</th>
                        <th>C&W MESSAGE TEXT</th>
                    </tr>
                </thead>
                <tbody>
                    {tableEntries.map((entry) => {
                        return(
                            <tr key={entry.eventNum} className={entry.cl === "W" ? "red" : "yellow"}>
                                <td>{entry.eventNum}</td>
                                <td>{entry.annun}</td>
                                <td>{entry.cl}</td>
                                <td>{entry.ack ? "X" : ""}</td>
                                <td>{entry.sys}</td>
                                <td>{entry.message}</td>
                            </tr>
                        )
                    })}
                </tbody>
            </Table>
        </div>
    )
};

const WarningSummary = styled(UWarningSummary)`
  min-height: 50%;
  background-color: rgba(57, 57, 57, 0.5);
  border-radius: 5px;
  
  table {
    table-layout: fixed;
    width: 100%;
    height: 100%;
    margin-top: 20px;
    color: black;
    line-height: 8px;
  }

  table td, table th {
      height: 25px;
      vertical-align: middle;
  }

  .white {
      color: white;
  }


  .yellow {
      background-color: #fcea5d;
  }

  .red {
    background-color: #f54242;
    color: white;
  }
`;

export default connector(WarningSummary);