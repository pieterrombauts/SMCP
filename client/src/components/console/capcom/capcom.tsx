import React from 'react';
import styled from 'styled-components';
import Nav from 'react-bootstrap/Nav'
import Row from 'react-bootstrap/Row'
import Col from 'react-bootstrap/Col'
import Tab from 'react-bootstrap/Tab'
import WarningSummary from './WarningSummary'
import ButtonsDisplay from './ButtonsDisplay'
import SignalDisplay from './SignalDisplay'

interface AppProps {
  className?: string;
};

const capcom: React.FC<AppProps> = (props) => {
  return (
    <>
      <WarningSummary />
      <SignalDisplay />
      {/* <ButtonsDisplay /> */}
    </>
  );
}

export const Capcom = styled(capcom)`
  width: 100%;
  height: 100%;
  position: absolute;
`;