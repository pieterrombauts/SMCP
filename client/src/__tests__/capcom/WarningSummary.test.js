import React from "react";
import 'jest-styled-components'
import toJson from 'enzyme-to-json'
import { WarningSummary } from '../../components/console/capcom/WarningSummary'
import { shallow, mount, render} from 'enzyme';

describe("Warning summary display", () => {
  it('renders correctly', () => {
    const wrapper = mount(<WarningSummary />)
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  })
})