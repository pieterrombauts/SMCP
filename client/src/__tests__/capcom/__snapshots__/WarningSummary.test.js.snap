// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Warning summary display renders correctly 1`] = `
.c0 {
  min-height: 50%;
  background-color: #FFF;
  border-radius: 5px;
}

.c0 table {
  table-layout: fixed;
  width: 100%;
  height: 100%;
  margin-top: 20px;
  color: black;
  line-height: 8px;
}

.c0 table td,
.c0 table th {
  height: 25px;
  vertical-align: middle;
}

.c0 .white {
  color: white;
}

.c0 .yellow {
  background-color: #fcea5d;
}

.c0 .red {
  background-color: #f54242;
  color: white;
}

<Styled(UWarningSummary)>
  <UWarningSummary
    className="c0"
  >
    <div
      className="c0"
    >
      <ForwardRef
        bordered={true}
        hover={true}
        size="sm"
        variant="dark"
      >
        <table
          className="table table-dark table-sm table-bordered table-hover"
        >
          <thead>
            <tr
              className="white"
            >
              <th
                style={
                  Object {
                    "width": "100px",
                  }
                }
              >
                EVENT#
              </th>
              <th
                style={
                  Object {
                    "width": "100px",
                  }
                }
              >
                ANNUN
              </th>
              <th
                style={
                  Object {
                    "width": "100px",
                  }
                }
              >
                CL
              </th>
              <th
                style={
                  Object {
                    "width": "100px",
                  }
                }
              >
                ACK
              </th>
              <th
                style={
                  Object {
                    "width": "100px",
                  }
                }
              >
                SYS
              </th>
              <th>
                C&W MESSAGE TEXT
              </th>
            </tr>
          </thead>
          <tbody>
            <tr
              className="yellow"
              key="9413"
            >
              <td>
                9413
              </td>
              <td>
                SUPP
              </td>
              <td>
                C
              </td>
              <td />
              <td>
                TCS
              </td>
              <td>
                Thermal Safing Load Shed Inhibited - LAB
              </td>
            </tr>
            <tr
              className="yellow"
              key="12288"
            >
              <td>
                12288
              </td>
              <td>
                SUPP
              </td>
              <td>
                C
              </td>
              <td />
              <td>
                TCS
              </td>
              <td>
                Thermal Thermal Safing Node 2 LTL Load Shed Timer Started - Node2
              </td>
            </tr>
            <tr
              className="yellow"
              key="13463"
            >
              <td>
                13463
              </td>
              <td>
                SUPP
              </td>
              <td>
                C
              </td>
              <td />
              <td>
                CDH
              </td>
              <td>
                DMS Mission Management Computer (MMC) Failure - COL
              </td>
            </tr>
            <tr
              className="yellow"
              key="13598"
            >
              <td>
                13598
              </td>
              <td>
                ENA
              </td>
              <td>
                C
              </td>
              <td />
              <td>
                EPS
              </td>
              <td>
                Power Distribution Unit (PDU2) Output Overcurrent Trip - COL
              </td>
            </tr>
            <tr
              className="yellow"
              key="13568"
            >
              <td>
                13568
              </td>
              <td>
                SUPP
              </td>
              <td>
                C
              </td>
              <td />
              <td>
                CDH
              </td>
              <td>
                DMS System Bus Failure - COL
              </td>
            </tr>
            <tr
              className="yellow"
              key="13578"
            >
              <td>
                13578
              </td>
              <td>
                ENA
              </td>
              <td>
                C
              </td>
              <td />
              <td>
                EPS
              </td>
              <td>
                Power Distribution Unit (PDU1) Nominal Controller Failure - COL
              </td>
            </tr>
            <tr
              className="yellow"
              key="12592"
            >
              <td>
                12592
              </td>
              <td>
                SUPP
              </td>
              <td>
                C
              </td>
              <td />
              <td>
                ECL
              </td>
              <td>
                Cabin Smoke Detector 2 Fail - COL
              </td>
            </tr>
            <tr
              className="yellow"
              key="5049"
            >
              <td>
                5049
              </td>
              <td>
                SUPP
              </td>
              <td>
                C
              </td>
              <td />
              <td>
                TCS
              </td>
              <td>
                Thermal Saffing Partial LTL Load Shed Timer Started
              </td>
            </tr>
            <tr
              className="yellow"
              key="5050"
            >
              <td>
                5050
              </td>
              <td>
                SUPP
              </td>
              <td>
                C
              </td>
              <td />
              <td>
                TCS
              </td>
              <td>
                Thermal Saffing Partial MTL Load Shed Timer Started
              </td>
            </tr>
            <tr
              className="yellow"
              key="13529"
            >
              <td>
                13529
              </td>
              <td>
                SUPP
              </td>
              <td>
                C
              </td>
              <td />
              <td>
                ECL
              </td>
              <td>
                ppO2 Sensor 2 Low - COL
              </td>
            </tr>
            <tr
              className="yellow"
              key="13558"
            >
              <td>
                13558
              </td>
              <td>
                SUPP
              </td>
              <td>
                C
              </td>
              <td />
              <td>
                ECL
              </td>
              <td>
                Loss of IMV Supply Function - COL
              </td>
            </tr>
            <tr
              className="yellow"
              key="4173"
            >
              <td>
                4173
              </td>
              <td>
                ENA
              </td>
              <td>
                C
              </td>
              <td />
              <td>
                TCS
              </td>
              <td>
                ETCS Loop B PCVP Loss of Comm - P1
              </td>
            </tr>
            <tr
              className="red"
              key="13535"
            >
              <td>
                13535
              </td>
              <td>
                SUPP
              </td>
              <td>
                W
              </td>
              <td />
              <td>
                TCS
              </td>
              <td>
                Cooling Loop Delta Pressure Sensor 2 Low - COL
              </td>
            </tr>
            <tr
              className="yellow"
              key="10472"
            >
              <td>
                10472
              </td>
              <td>
                ENA
              </td>
              <td>
                C
              </td>
              <td>
                X
              </td>
              <td>
                TCS
              </td>
              <td>
                ETCS Loop B Radiator Environment Temp Too Cold - P1
              </td>
            </tr>
            <tr
              className="yellow"
              key="4248"
            >
              <td>
                4248
              </td>
              <td>
                ENA
              </td>
              <td>
                C
              </td>
              <td>
                X
              </td>
              <td>
                TCS
              </td>
              <td>
                ETCS Loop B TRRJ Switchover Sequence Failure - P1
              </td>
            </tr>
            <tr
              className="yellow"
              key="5100"
            >
              <td>
                5100
              </td>
              <td>
                SUPP
              </td>
              <td>
                C
              </td>
              <td>
                X
              </td>
              <td>
                CNT
              </td>
              <td>
                Loss of Active IAC Handover to Backup IAC
              </td>
            </tr>
            <tr
              className="yellow"
              key="11099"
            >
              <td>
                11099
              </td>
              <td>
                ENA
              </td>
              <td>
                C
              </td>
              <td>
                X
              </td>
              <td>
                EPS
              </td>
              <td>
                RPCM S32B_A Observed vs Last Commanded State Discrepancy - S3
              </td>
            </tr>
          </tbody>
        </table>
      </ForwardRef>
    </div>
  </UWarningSummary>
</Styled(UWarningSummary)>
`;
